app-frontend/src/types/initialState.ts

```diff
@@ -0,0 +1,30 @@
+import { Link, SearchedLink } from './link';
+import { UserInterface } from './user';
+
+export interface InitialState {
```

Comment:
the initial state should be combined from individual initial states of reducers. 
In order, initial state of reduced should be instantiated in appropriate reducers